src/test/java/uk/gov/ons/ctp/response/collection/exercise/client/ActionSvcClientTest.java

```diff
@@ -105,6 +110,85 @@ public void createActionPlan_500Response() {
         .postForObject(any(URI.class), any(HttpEntity.class), eq(ActionPlanDTO.class));
   }
 
+  @Test
+  public void updateActionPlanNameAndDescription_200Response() throws CTPException {
+
+    // Given
+    when(restTemplate.exchange(
+            any(URI.class), eq(HttpMethod.PUT), any(HttpEntity.class), eq(ActionPlanDTO.class)))
+        .thenReturn(actionPlanResponseEntity);
+
+    // When
+    ActionPlanDTO actionPlan =
+        actionSvcRestClient.updateActionPlanNameAndDescription(
+            UUID.fromString(ACTION_PLAN_ID), "test_name", "test_description");
+
+    assertEquals(actionPlan.getId(), actionPlans.get(0).getId());
+  }
+
+  @Test(expected = HttpClientErrorException.class)
+  public void updateActionPlanNameAndDescription_500Response() {
+    // Given
+    when(restTemplate.exchange(
+            any(URI.class), eq(HttpMethod.PUT), any(HttpEntity.class), eq(ActionPlanDTO.class)))
+        .thenThrow(new HttpClientErrorException(HttpStatus.INTERNAL_SERVER_ERROR));
```

Comment:
a client error wont have a 500 response only 400s. An internal server error will throw a HttpServerErrorException not a HttpClientErrorException

Reply:
done!